content/2016-04-18-aliasing-improved.rst

```diff
@@ -0,0 +1,473 @@
+
+.. code:: python
+
+    >>> for i in range(1000):
+    ...     pass
+
+identifier ``i`` can be bound to a great deal of values, and we cannot track them individually.
+Instead Pythran only keep tracks of values that are bound to an identifier. All
+the others are hidden between the terms of ``<unbound-value>``.
+
+So let's start to write some simple equations [1]_, with a few test cases demonstrated as Python code which needs some initialization:
+
+.. code::
+
+    >>> import ast
+    >>> from pythran.analyses.aliases import *
+    >>> from pythran import passmanager
+    >>> pm = passmanager.PassManager('demo')
+
+Here we basically inject the ``aliases`` namespace into current namespace for
+convenience, then create an instance of the object in charge of applying passes
+and gathering analyse results.
+
+Bool Op Expression
+------------------
+
+*(A.k.a ``or`` and ``and``)*
+
+Resulting node may alias to either operands:
+
+.. code:: python
+
+    >>> module = ast.parse('def foo(a, b): return a or b')
+    >>> result = pm.gather(Aliases, module)
+    >>> Aliases.dump(result, filter=ast.BoolOp)
+    (a or b) => ['a', 'b']
+
+This code snippet requires a few explanations:
+
+    #. First, it parses a code snippet and turns it into an Abstract Syntax Tree (AST).
+
+    #. Second, it computes the alias information at every point of the program.
+           ``result`` is a dictionary that maps nodes from the AST to set of
+           identifiers (remember that for Pythran, a node can only alias to
+           bounded values. These values are represented by the first identifier
+           they are bound to).
+
+    #. Finally, it pretty prints the result of the analysis, using a filter to
+           only dump the part we are interested in. In that case it dumps a
+           textual representation of the alias set of the ``ast.BoolOp`` nodes,
+           which turns out to be ``['a', 'b']``.
+
+Unary Operator Expression
+-------------------------
+
+Resulting node does not alias to anything
+
+.. code:: python
+
+    >>> module = ast.parse('def foo(a): return -a')
+    >>> result = pm.gather(Aliases, module)
+    >>> Aliases.dump(result, filter=ast.UnaryOp)
+    (- a) => ['<unbound-value>']
+
+As stated previously, values not bound to an identifier are only represented as ``<unbound-value>``.
+
+If Expression
+-------------
+
+Resulting node alias to either branch
+
+.. code:: python
+
+    >>> module = ast.parse('def foo(a, b, c): return a if c else b')
+    >>> result = pm.gather(Aliases, module)
+    >>> Aliases.dump(result, filter=ast.IfExp)
+    (a if c else b) => ['a', 'b']
+
+Dict Expression
+---------------
+
+A dict is abstracted as an unordered container of its values
+
+.. code:: python
+
+    >>> module = ast.parse('def foo(a, b): return {0: a, 1: b}')
+    >>> result = pm.gather(Aliases, module)
+    >>> Aliases.dump(result, filter=ast.Dict)
+    {0: a, 1: b} => ['|a|', '|b|']
```

Comment:
Not yet :-) But there is no reason we could not handle string literals as well,
you're right